controllers/sdkutil/clientwrapper/autoscaling_client.go

```diff
@@ -0,0 +1,162 @@
+type applicationAutoscalingClientWrapper struct {
+	ApplicationAutoscalingClientWrapper
+	innerClient applicationautoscalingiface.ClientAPI
+}
+
+// RegisterScalableTarget registers a scalable target. Returns the response output or nil if error.
+func (c *applicationAutoscalingClientWrapper) RegisterScalableTarget(ctx context.Context, autoscalingTarget *applicationautoscaling.RegisterScalableTargetInput) (*applicationautoscaling.RegisterScalableTargetOutput, error) {
+
+	createRequest := c.innerClient.RegisterScalableTargetRequest(autoscalingTarget)
+	response, err := createRequest.Send(ctx)
+
+	if response != nil {
+		return response.RegisterScalableTargetOutput, nil
+	}
+	return nil, err
+}
+
+// RegisterScalableTarget registers a scalable target. Returns the response output or nil if error.
+func (c *applicationAutoscalingClientWrapper) PutScalingPolicy(ctx context.Context, autoscalingJob *applicationautoscaling.PutScalingPolicyInput) (*applicationautoscaling.PutScalingPolicyOutput, error) {
+
+	createRequest := c.innerClient.PutScalingPolicyRequest(autoscalingJob)
+	response, err := createRequest.Send(ctx)
+
+	if response != nil {
+		return response.PutScalingPolicyOutput, nil
+	}
+
+	return nil, err
+}
+
+// DeleteScalingPolicy Deletes the scaling policy
+func (c *applicationAutoscalingClientWrapper) DeleteScalingPolicy(ctx context.Context, autoscalingJob *applicationautoscaling.DeleteScalingPolicyInput) (*applicationautoscaling.DeleteScalingPolicyOutput, error) {
+	deleteRequest := c.innerClient.DeleteScalingPolicyRequest(autoscalingJob)
+	deleteResponse, deleteError := deleteRequest.Send(ctx)
+
+	if deleteError != nil {
+		return nil, deleteError
+	}
+
+	return deleteResponse.DeleteScalingPolicyOutput, deleteError
+}
+
+// DeregisterScalableTarget deregisters a scalable target
+func (c *applicationAutoscalingClientWrapper) DeregisterScalableTarget(ctx context.Context, autoscalingJob *applicationautoscaling.DeregisterScalableTargetInput) (*applicationautoscaling.DeregisterScalableTargetOutput, error) {
+	deleteRequest := c.innerClient.DeregisterScalableTargetRequest(autoscalingJob)
+	deleteResponse, deleteError := deleteRequest.Send(ctx)
+
+	if deleteError != nil {
+		return nil, deleteError
+	}
+
+	return deleteResponse.DeregisterScalableTargetOutput, deleteError
+}
+
+// DescribeScalableTargets returns the scalableTarget description filtered on PolicyName and a single ResourceID
+// TODO: change this to return only the ScalableTargetObject for cleaner descriptions
+func (c *applicationAutoscalingClientWrapper) DescribeScalableTargets(ctx context.Context, resourceID string) (*applicationautoscaling.DescribeScalableTargetsOutput, error) {
+
+	var resourceIDList []string
+	resourceIDList = append(resourceIDList, resourceID)
+	// Review: This filtered response should be of size 1 by default
+	var maxResults int64 = 1
+
+	// TODO: Remove hardcoded values, might need to construct the input object
```

Comment:
Not really sure how you would reduce this much, but yeah this repeated code seems annoying.

Reply:
yea, letting it be for now